### PR TITLE
[Transformer] Skip dropout layer when rate=0

### PR DESCRIPTION
**Pull request by TaoLv**
## Description ##
https://github.com/dmlc/gluon-nlp/issues/596

### Essentials ###
- [ ] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by mli** [comment]
Job PR-597/1 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-597/1/index.html

File: src/gluonnlp/model/transformer.py [hunk 1]
```diff
@@ -106,7 +107,8 @@ def __init__(self, units=512, hidden_size=2048, dropout=0.0, use_residual=True,
                                   weight_initializer=weight_initializer,
                                   bias_initializer=bias_initializer,
                                   prefix='ffn_2_')
-            self.dropout_layer = nn.Dropout(dropout)
```
Comment (eric-haibin-lin) [inline src/gluonnlp/model/transformer.py]: Might as well change 
https://github.com/dmlc/gluon-nlp/blob/972d8666b287e0e58a6273adf186307b79f2e2ba/src/gluonnlp/model/transformer.py#L207
https://github.com/dmlc/gluon-nlp/blob/972d8666b287e0e58a6273adf186307b79f2e2ba/src/gluonnlp/model/transformer.py#L339
https://github.com/dmlc/gluon-nlp/blob/972d8666b287e0e58a6273adf186307b79f2e2ba/src/gluonnlp/model/transformer.py#L719
and 
https://github.com/dmlc/gluon-nlp/blob/972d8666b287e0e58a6273adf186307b79f2e2ba/src/gluonnlp/model/transformer.py#L854
?

Comment (TaoLv) [reply src/gluonnlp/model/transformer.py]: Sure. Will change them accordingly.

**Comment by szhengac** [comment]
Looks ok to me.

**Comment by TaoLv** [comment]
Looks like the failure in CI is not related to the changes in this PR. Any idea about how to get it passed?

**Comment by szha** [comment]
@TaoLv I triggered the CI again just now. I will make the linkcheck optional and instead let @mli send a report to the PR if anything is broken

File: src/gluonnlp/model/transformer.py (hunk 1 above)
Comment (eric-haibin-lin) [reply src/gluonnlp/model/transformer.py]: Thanks. Would you mind also adding the case when dropout is set to 0 in the unit test? https://github.com/dmlc/gluon-nlp/blob/master/tests/unittest/test_models.py#L75

**Comment by TaoLv** [comment]
Sure, will change accordingly.

**Comment by mli** [comment]
Job PR-597/5 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-597/5/index.html

**Comment by mli** [comment]
Job PR-597/6 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-597/6/index.html
